subprojects/model-core/src/main/java/org/gradle/internal/credentials/DefaultHttpHeaderCredentials.java

```diff
@@ -16,12 +16,14 @@
 package org.gradle.internal.credentials;
 
 import org.gradle.api.credentials.HttpHeaderCredentials;
+import org.gradle.api.tasks.Internal;
 
 public class DefaultHttpHeaderCredentials implements HttpHeaderCredentials {
 
     private String name;
     private String value;
 
+    @Internal
```

Comment:
Is the header name confidential? It probably isn't, so I would mark it as `@Input`. Note that the header name can be null, so you also need `@Optional`.